File: optimization/loss.cpp
double* b;
int arraySize;
//calculate FFT of sample
double* bFFT;

double loss(double* a) {
	//calculate synth output with parameters in a
	//calculate FFT of the synth output
	double* aFFT;
	double errorSUM = 0;
	for (int i = 0; i < arraySize; i++) {
		errorSUM += (a[i] - b[i]) * (a[i] - b[i]);
	}
	return errorSUM / arraySize;
}

File: optimization/loss.py
import scipy
import numpy
import math

def loss(parameters):
	#call synth function with parameters and store value into x
	x = []
	xFFT = scipy.fft.fft(x)
	y = data = numpy.genfromtxt('targetSound.csv', delimiter=",")
	yFFT = scipy.fft.fft(y)
	sum = 0.0
	sumI = 0.0j
	divideVal = 0
	for i in range(math.min(xFFT.size, yFFT.size)):
		sum += math.pow(xFFT[i].real - yFFT[i].real, 2)
		sumI += math.pow(xFFT[i].imag - yFFT[i].imag, 2)
		divideVal += 1
	return (sum + sumI.real)/divideVal